packages/datadog-instrumentations/src/cassandra-driver.js

```diff
@@ -0,0 +1,207 @@
+'use strict'
+
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+
+addHook({ name: 'cassandra-driver', versions: ['>=3.0.0'] }, cassandra => {
+  const startCh = channel('apm:cassandra:batch:start')
+  const asyncEndCh = channel('apm:cassandra:batch:async-end')
+  const endCh = channel('apm:cassandra:batch:end')
+  const errorCh = channel('apm:cassandra:batch:error')
+
+  shimmer.wrap(cassandra.Client.prototype, 'batch', batch => function (queries, options, callback) {
+    const asyncResource = new AsyncResource('bound-anonymous-fn')
+    const query = combine(queries)
```

Comment:
The individual queries should be published as-is, it should be the subscriber that decides to combine them.

Reply:
Alternatively, depending on how the driver works, it could be combined but then published with a `query` event. This only makes sense if the entire batch is sent as one query to the server.